github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/dsl/GitHubPRJobDslExtension.java

```diff
@@ -44,4 +46,18 @@ public Object commitStatusOnGH(Runnable closure) throws ANTLRException {
                 null
         );
     }
+
+    @DslExtensionMethod(context = StepContext.class)
+    public Object updateStatusOnGH(Runnable closure) {
+        GitHubPRStatusStepDslContext context = new GitHubPRStatusStepDslContext();
+        executeInContext(closure, context);
+
+        return new GitHubPRBuildStatusPublisher(
```

Comment:
Shall I create `GitHubPRStatusBuilder ` instead of using `GitHubPRBuildStatusPublisher`?

I didn't do it because I'd be just a copy paste job with very little value. The reality is that I want to use the publisher in the Builder/Step context.
In other words, it's not a proper builder (it doesn't build anything). It just publishes a message.